autonav_ws/src/autonav_shared/autonav_shared/node.py

```diff
@@ -0,0 +1,235 @@
+        """
+        try:
+            result = future.result()
+            if result is None or not result.ok:
+                self.log("Failed to set system state", LogLevel.ERROR)
+            else:
+                self.log("Successfully set system state", LogLevel.DEBUG)
+        except Exception as e:
+            self.log(f"Failed to set system state: {e}", LogLevel.ERROR)
+
+    def get_device_state(self, device: str = None) -> DeviceState:
+        """
+        Get the state of a specific device.
+
+        If no device is provided, the state of the current node is returned.
+        """
+        if device is None:
+            device = self.get_name()
+
+        if device not in self.device_states:
+            self.log(f"Device {device} not found in device states", LogLevel.ERROR)
+            return DeviceState.UNKNOWN
+        
+        return self.device_states[device]
+    
+    def get_system_state(self) -> SystemState:
+        """
+        Get the state of the system.
+        """
+        return self.system_state
+    
+    def is_mobility(self) -> bool:
+        """
+        Check if the system is in mobility mode.
+        """
+        return self.mobility
+
+    def log(self, message: str, level: LogLevel = LogLevel.INFO) -> None:
+        """
+        Log a message with a given log level.
+        """
+        # Get the current time as yyyy-mm-dd hh:mm:ss:ms
+        current_time = time.strftime("%Y-%m-%d %H:%M:%S:", time.localtime()) + str(time.time() % 1)[2:5]
+        
+        # Get the calling function name and line number
+        frame = inspect.currentframe().f_back
+        calling_function = frame.f_code.co_name
+        line_number = frame.f_lineno
+        
+        # Get the log level as a string
+        level_str = LogLevel(level).name
+        level_str = level_str + " " * (5 - len(level_str))
+
+        # Log to topic
+        msg = Log()
+        msg.timestamp = time.time_ns() // 1_000_000
+        msg.level = level.value
+        msg.node = self.get_name()
+        msg.function_caller = calling_function
+        msg.line_number = line_number
+        msg.message = message
+        self.log_pub.publish(msg)
+        
+        match level:
+            case LogLevel.DEBUG: # 61, 117, 157
+                print(f"{sty.fg(99, 150, 79)}{current_time} {sty.fg.white}| {sty.fg(61, 117,157)}{level_str} {sty.fg.white}| {sty.fg(90, 60, 146)}{self.get_name()}{sty.fg.white}:{sty.fg(90, 60, 146)}{calling_function}{sty.fg.white}:{sty.fg(90, 60, 146)}{line_number} {sty.fg.white}- {sty.fg(61, 117, 157)}{message}{sty.rs.all}")
+                
+            case LogLevel.INFO: # 255, 255, 255
+                print(f"{sty.fg(99, 150, 79)}{current_time} {sty.fg.white}| {sty.fg(255, 255, 255)}{level_str} {sty.fg.white}| {sty.fg(90, 60, 146)}{self.get_name()}{sty.fg.white}:{sty.fg(90, 60, 146)}{calling_function}{sty.fg.white}:{sty.fg(90, 60, 146)}{line_number} {sty.fg.white}- {sty.fg(255, 255, 255)}{message}{sty.rs.all}")
```

Comment:
it's fine because it works but I don't like super repetitive code, what if the switch statement set a variable for the color and then you had only 1 print statement after the switch statement